crates/configuration/src/parachain.rs

```diff
@@ -1,41 +1,47 @@
 use std::{cell::RefCell, error::Error, fmt::Display, marker::PhantomData, rc::Rc};
 
 use multiaddr::Multiaddr;
+use serde::Serialize;
 
 use crate::shared::{
     errors::{ConfigError, FieldError},
     helpers::{merge_errors, merge_errors_vecs},
     macros::states,
     node::{self, NodeConfig, NodeConfigBuilder},
     resources::{Resources, ResourcesBuilder},
-    types::{Arg, AssetLocation, Chain, ChainDefaultContext, Command, Image, ValidationContext},
+    types::{
+        Arg, AssetLocation, Chain, ChainDefaultContext, Command, Image, ValidationContext, U128,
+    },
 };
 
-#[derive(Debug, Clone, PartialEq)]
+#[derive(Debug, Clone, PartialEq, Serialize)]
 pub enum RegistrationStrategy {
     InGenesis,
     UsingExtrinsic,
 }
 
 /// A parachain configuration, composed of collators and fine-grained configuration options.
-#[derive(Debug, Clone, PartialEq)]
+#[derive(Debug, Clone, PartialEq, Serialize)]
 pub struct ParachainConfig {
     id: u32,
     chain: Option<Chain>,
     registration_strategy: Option<RegistrationStrategy>,
-    initial_balance: u128,
+    initial_balance: U128,
     default_command: Option<Command>,
     default_image: Option<Image>,
     default_resources: Option<Resources>,
     default_db_snapshot: Option<AssetLocation>,
+    #[serde(skip_serializing_if = "std::vec::Vec::is_empty")]
     default_args: Vec<Arg>,
     genesis_wasm_path: Option<AssetLocation>,
     genesis_wasm_generator: Option<Command>,
     genesis_state_path: Option<AssetLocation>,
     genesis_state_generator: Option<Command>,
     chain_spec_path: Option<AssetLocation>,
     is_cumulus_based: bool,
```

Comment:
```suggestion
    // TODO: parse both case in zombienet node version to avoid renamed ?
    #[serde(rename = "cumulus_based")]
    is_cumulus_based: bool,
```